Orange/widgets/data/owrank.py

```diff
@@ -142,14 +142,17 @@ def headerData(self, section, orientation, role=Qt.DisplayRole):
 
     def setExtremesFrom(self, column, values):
         """Set extremes for columnn's ratio bars from values"""
-        try:
-            vmin = np.nanmin(values)
-            if np.isnan(vmin):
-                raise TypeError
-        except TypeError:
-            vmin, vmax = -np.inf, np.inf
-        else:
-            vmax = np.nanmax(values)
+        with warnings.catch_warnings():
+            warnings.filterwarnings("ignore", ".*All-NaN slice encountered.*",
+                                    RuntimeWarning)
+            try:
+                vmin = np.nanmin(values)
+                if np.isnan(vmin):
+                    raise TypeError
+            except TypeError:
```

Comment:
I prefer explicit code that checks before doing something "stupid" instead of doing it first and catching exceptions.
If we had that here, we would also not need to suppress warnings...

I am guessing it would be enough to check `if np.isfinite(values).any()` and use the computed values, otherwise set to inf?
Shorter, clearer, explicit.
(and not tested, so maybe not enough :))

Sidenote: If sometime in the future, we pass a list of values (instead of np.array), the warning would change from
All-NaN slice encountered
to
All-NaN axis encountered
Just to keep in mind when setting this filter in functions that are otherwise fine with any iterable.

Reply:
I agree but we don't understand the code well enough to dare to remove catching the `TypeError`. Hence I only changed the code to filter only warnings only in `vmin = np.nanmin(values)`.